src/Components/Server/src/ComponentHub.cs

```diff
@@ -72,7 +72,7 @@ public override Task OnDisconnectedAsync(Exception exception)
             return _circuitRegistry.DisconnectAsync(circuitHost, Context.ConnectionId);
         }
 
-        public async ValueTask<string> StartCircuit(string baseUri, string uri)
+        public async ValueTask<string> StartCircuit(string baseUri, string uri, string serializedDescriptors)
```

Comment:
We can change this in the future to take just the first component and add an additional hub method `RenderComponent` where we render components after the first one. That will better support scenarios with arguments.

Reply:
Is the idea right now that you run into a physical limit based on the size of this message (24kb I think?). Do you know how many components it would take to reach that? obvi depends on the length of names, but can you estimate something reasonable and try this?

Reply:
The default message size is 32Kb, each component is about 500 bytes based on just looking at the length in the startup method, so you can render about 65 components without parameters. This size might be reduced a bit, but estimate that between 100 and 200 tops, and that's without parameters.